src/NUnitEngine/nunit.engine/Services/TestSelectionParser.cs

```diff
@@ -125,7 +125,9 @@ public string ParseFilterElement()
             {
                 case "test":
                     op = Expect(REL_OPS);
-                    rhs = GetTestName();
+                    rhs = op == MATCH_OP || op == NOMATCH_OP
+                        ? Expect(TokenKind.String, TokenKind.Word)
+                        : GetTestName();
```

Comment:
All tests pass without this change. Could you either remove it or add a test that shows it's required?

Reply:
Ah, of course! A regex __may__ be an actual test name, but is only required to __match__ a test name. So calling `TestTestName` really doesn't make sense. This means that the user has to be especially precise in formulating regular expressions and also has to be aware of how best to enter them at the command-line for each OS. But it has always been so with regex. :-)

No change needed. I'm going to merge this.